src/solvers/refinement/string_builtin_function.cpp

```diff
@@ -187,6 +187,46 @@ exprt string_set_char_builtin_functiont::length_constraint() const
     equal_exprt(return_code, return_value));
 }
 
+static bool eval_is_upper_case(const mp_integer &c)
+{
+  return ('A' <= c && c <= 'Z') || (0xc0 <= c && c <= 0xd6) ||
+         (0xd8 <= c && c <= 0xde);
+}
+
+optionalt<exprt> string_to_lower_case_builtin_functiont::eval(
```

Comment:
I'm probably not paying attention, but why not just return the exprt anyway?  This is how the rewriter works.

Reply:
If the eval function is not successful, the solver will get the valuation in another way (calling get of the underlying solver and doing some post-processing).

Reply:
OK... is this called before or after solving?

Reply:
It's called from `get` at the end of each string refinement step after solving by the underlying solver.